CHANGELOG for 2.19.0 (#7829)
timescale/timescaledb: CHANGELOG.md

`psql` with the `-X` flag to prevent any `.psqlrc` commands from
accidentally triggering the load of a previous DB version.**

## 2.19.0 (2025-03-18)

This release contains performance improvements and bug fixes since  the 2.18.2 release. We recommend that you upgrade at the next  available opportunity.

> 

Author: ```suggestion
**Highlights of the v2.19.0 release**
* The Hypercore columnstore no longer blocks DML statements during recompression, allowing INSERT, UPDATE, and DELETE operations on compressed chunks significantly improves concurrency.
* Improvements to Continuous Aggregation now materialize the most recent data first, to enhance query experience. Additionally, the refresh policy now supports batching, which helps reduce system pressure and minimize the risk of spilling to disk, leading to better overall performance.
* We are further investing in vectorization (SIMD) to enhance Hypercore’s performance, improving analytical queries such as aggregations over text columns and those with an arbitrary number of columns to aggregate on.
```

* [#7790](https://github.com/timescale/timescaledb/pull/7790) Add configurable Incremental CAgg Refresh Policy

**Bugfixes**
* [#7665](https://github.com/timescale/timescaledb/pull/7665) Block merging of frozen chunks
* [#7673](https://github.com/timescale/timescaledb/pull/7673) Don't abort additional INSERTs when hitting first conflict
* [#7714](https://github.com/timescale/timescaledb/pull/7714) Fixes a wrong result when compressed NULL values were confused with default values. This happened in very special circumstances with alter table added a new column with a default value, an update and compression in a very particular order.
* [#7747](https://github.com/timescale/timescaledb/pull/7747) Block TAM rewrites with incompatible GUC setting
* [#7748](https://github.com/timescale/timescaledb/pull/7748) Crash in the segmentwise recompression
* [#7764](https://github.com/timescale/timescaledb/pull/7764) Fix compression settings handling in Hypercore TAM
* [#7768](https://github.com/timescale/timescaledb/pull/7768) Remove costing index scan of hypertable parent
* [#7799](https://github.com/timescale/timescaledb/pull/7799) Handle DEFAULT table access name in ALTER TABLE

> 

Author: ```suggestion
**GUCs**
* TBD
```

`psql` with the `-X` flag to prevent any `.psqlrc` commands from
accidentally triggering the load of a previous DB version.**

## 2.19.0 (2025-03-18)

This release contains performance improvements and bug fixes since  the 2.18.2 release. We recommend that you upgrade at the next  available opportunity.
**Highlights of the v2.19.0 release**
* The Hypercore columnstore no longer blocks DML statements during recompression, allowing INSERT, UPDATE, and DELETE operations on compressed chunks significantly improves concurrency.
* Improvements to Continuous Aggregation now materialize the most recent data first, to enhance query experience. Additionally, the refresh policy now supports batching, which helps reduce system pressure and minimize the risk of spilling to disk, leading to better overall performance.
* We are further investing in vectorization (SIMD) to enhance Hypercore’s performance, improving analytical queries such as aggregations over text columns and those with an arbitrary number of columns to aggregate on.

> **Features**

Author: ```suggestion

**Deprecation warning**

This is the last minor release supporting PostgreSQL 14. Starting with the minor version of TimescaleDB only Postgres 15, 16 and 17 will be supported.

**Features**
```

* [#7790](https://github.com/timescale/timescaledb/pull/7790) Add configurable Incremental CAgg Refresh Policy

**Bugfixes**
* [#7665](https://github.com/timescale/timescaledb/pull/7665) Block merging of frozen chunks
* [#7673](https://github.com/timescale/timescaledb/pull/7673) Don't abort additional INSERTs when hitting first conflict
* [#7714](https://github.com/timescale/timescaledb/pull/7714) Fixes a wrong result when compressed NULL values were confused with default values. This happened in very special circumstances with alter table added a new column with a default value, an update and compression in a very particular order.
* [#7747](https://github.com/timescale/timescaledb/pull/7747) Block TAM rewrites with incompatible GUC setting
* [#7748](https://github.com/timescale/timescaledb/pull/7748) Crash in the segmentwise recompression
* [#7764](https://github.com/timescale/timescaledb/pull/7764) Fix compression settings handling in Hypercore TAM
* [#7768](https://github.com/timescale/timescaledb/pull/7768) Remove costing index scan of hypertable parent
* [#7799](https://github.com/timescale/timescaledb/pull/7799) Handle DEFAULT table access name in ALTER TABLE

> **GUCs**

Author: ```suggestion

**GUCs**
```

accidentally triggering the load of a previous DB version.**

## 2.19.0 (2025-03-18)

This release contains performance improvements and bug fixes since  the 2.18.2 release. We recommend that you upgrade at the next  available opportunity.
**Highlights of the v2.19.0 release**
* The Hypercore columnstore no longer blocks DML statements during recompression, allowing INSERT, UPDATE, and DELETE operations on compressed chunks significantly improves concurrency.
* Improvements to Continuous Aggregation now materialize the most recent data first, to enhance query experience. Additionally, the refresh policy now supports batching, which helps reduce system pressure and minimize the risk of spilling to disk, leading to better overall performance.
* We are further investing in vectorization (SIMD) to enhance Hypercore’s performance, improving analytical queries such as aggregations over text columns and those with an arbitrary number of columns to aggregate on.

**Deprecation warning**

This is the last minor release supporting PostgreSQL 14. Starting with the minor version of TimescaleDB only Postgres 15, 16 and 17 will be supported.

> 

Author: ```suggestion

**Downgrading from 2.19.0**

This release introduces custom bool compression, if you enable this feature via the `enable_bool_compression` and must downgrade to a previous, please use the [following script](https://github.com/timescale/timescaledb-extras/blob/master/utils/2.19.0-downgrade_new_compression_algorithms.sql) to convert the columns back to their previous state. TimescaleDB versions prior to 2.19.0 do not know how to handle this new type.

```

* [#7790](https://github.com/timescale/timescaledb/pull/7790) Add configurable Incremental CAgg Refresh Policy

**Bugfixes**
* [#7665](https://github.com/timescale/timescaledb/pull/7665) Block merging of frozen chunks
* [#7673](https://github.com/timescale/timescaledb/pull/7673) Don't abort additional INSERTs when hitting first conflict
* [#7714](https://github.com/timescale/timescaledb/pull/7714) Fixes a wrong result when compressed NULL values were confused with default values. This happened in very special circumstances with alter table added a new column with a default value, an update and compression in a very particular order.
* [#7747](https://github.com/timescale/timescaledb/pull/7747) Block TAM rewrites with incompatible GUC setting
* [#7748](https://github.com/timescale/timescaledb/pull/7748) Crash in the segmentwise recompression
* [#7764](https://github.com/timescale/timescaledb/pull/7764) Fix compression settings handling in Hypercore TAM
* [#7768](https://github.com/timescale/timescaledb/pull/7768) Remove costing index scan of hypertable parent
* [#7799](https://github.com/timescale/timescaledb/pull/7799) Handle DEFAULT table access name in ALTER TABLE

**GUCs**

> * TBD

Author: ```suggestion
* enable_bool_compression: enable the BOOL compression algorithm, default: OFF
```

* [#7790](https://github.com/timescale/timescaledb/pull/7790) Add configurable Incremental CAgg Refresh Policy

**Bugfixes**
* [#7665](https://github.com/timescale/timescaledb/pull/7665) Block merging of frozen chunks
* [#7673](https://github.com/timescale/timescaledb/pull/7673) Don't abort additional INSERTs when hitting first conflict
* [#7714](https://github.com/timescale/timescaledb/pull/7714) Fixes a wrong result when compressed NULL values were confused with default values. This happened in very special circumstances with alter table added a new column with a default value, an update and compression in a very particular order.
* [#7747](https://github.com/timescale/timescaledb/pull/7747) Block TAM rewrites with incompatible GUC setting
* [#7748](https://github.com/timescale/timescaledb/pull/7748) Crash in the segmentwise recompression
* [#7764](https://github.com/timescale/timescaledb/pull/7764) Fix compression settings handling in Hypercore TAM
* [#7768](https://github.com/timescale/timescaledb/pull/7768) Remove costing index scan of hypertable parent
* [#7799](https://github.com/timescale/timescaledb/pull/7799) Handle DEFAULT table access name in ALTER TABLE

**GUCs**

> * enable_bool_compression: enable the BOOL compression algorithm, default: OFF

Author: ```suggestion
* enable_bool_compression: enable the BOOL compression algorithm, default: OFF
* `enable_exclusive_locking_recompression`: enable exclusive locking during recompression (legacy mode), default: `OFF`
```

`psql` with the `-X` flag to prevent any `.psqlrc` commands from
accidentally triggering the load of a previous DB version.**

## 2.19.0 (2025-03-18)

This release contains performance improvements and bug fixes since  the 2.18.2 release. We recommend that you upgrade at the next  available opportunity.
**Highlights of the v2.19.0 release**

> * The Hypercore columnstore no longer blocks DML statements during recompression, allowing INSERT, UPDATE, and DELETE operations on compressed chunks significantly improves concurrency.
> * Improvements to Continuous Aggregation now materialize the most recent data first, to enhance query experience. Additionally, the refresh policy now supports batching, which helps reduce system pressure and minimize the risk of spilling to disk, leading to better overall performance.
> * We are further investing in vectorization (SIMD) to enhance Hypercore’s performance, improving analytical queries such as aggregations over text columns and those with an arbitrary number of columns to aggregate on.

Reviewer: ```suggestion
* Improved concurrency of INSERT, UPDATE and DELETE operations on the columnstore by no longer blocking DML statements during the recompression of a chunk.
* Improved system performance during Continuous Aggregates refreshes by breaking them into smaller batches which reduces systems pressure and minimizes the risk of spilling to disk.
* Faster and more up-to-date results for queries against Continuous Aggregates by materializing the most recent data first (vs old data first in prior versions).
* Faster analytical queries with SIMD vectorization of aggregations over text columns and group by over multiple column
* Enable optimizing chunk size for faster query performance on the columnstore by adding support for merging columnstore chunks to the merge_chunk API.n.
```

`psql` with the `-X` flag to prevent any `.psqlrc` commands from
accidentally triggering the load of a previous DB version.**

## 2.19.0 (2025-03-18)

This release contains performance improvements and bug fixes since  the 2.18.2 release. We recommend that you upgrade at the next  available opportunity.
**Highlights of the v2.19.0 release**
* Improved concurrency of INSERT, UPDATE and DELETE operations on the columnstore by no longer blocking DML statements during the recompression of a chunk.
* Improved system performance during Continuous Aggregates refreshes by breaking them into smaller batches which reduces systems pressure and minimizes the risk of spilling to disk.
* Faster and more up-to-date results for queries against Continuous Aggregates by materializing the most recent data first (vs old data first in prior versions).
* Faster analytical queries with SIMD vectorization of aggregations over text columns and group by over multiple column

> * Enable optimizing chunk size for faster query performance on the columnstore by adding support for merging columnstore chunks to the merge_chunk API.n.

Author: ```suggestion
* Enable optimizing chunk size for faster query performance on the columnstore by adding support for merging columnstore chunks to the `merge_chunk` API.
```
